### PR TITLE
Make TTL configurable

### PR DESCRIPTION
**Pull request by DarthMax**
~~Based on #719~~

### CODE REVIEW

**Comment by DarthMax** [comment]
[[🍓] Enable client to create sessions against self-managed dbms](https://trello.com/c/8FRF05Jk/9295-%F0%9F%8D%93-enable-client-to-create-sessions-against-self-managed-dbms)

File: doc/modules/ROOT/pages/gds-session.adoc [hunk 1]
```diff
@@ -61,6 +61,9 @@ You can use the `sessions.estimate()` method to estimate the size required.
 - **DBMS connection**.
 This is a `DbmsConnectionInfo` object that contains the URI of an Neo4j instance, a username, and a password.
 
+- **TTL**.
+This optional parameter specifies the time-to-live of the session. The session will be automatically deleted after the TTL has expired after the last user activity.
```
Comment (FlorentinD) [inline doc/modules/ROOT/pages/gds-session.adoc:65]:
```suggestion
This optional parameter specifies the time-to-live of the session. The session will be automatically deleted if the session was unused for the provided duration.
```

File: doc/modules/ROOT/pages/gds-session.adoc [hunk 2]
```diff
@@ -83,6 +87,7 @@ gds = sessions.get_or_create(
     session_name=name,
     memory=memory,
     db_connection=db_connection_info,
+    ttl=timedelta(hours=5),
```
Comment (FlorentinD) [inline doc/modules/ROOT/pages/gds-session.adoc:90]: how about we also add it to the notebook?

File: graphdatascience/session/aura_api.py [hunk 3]
```diff
@@ -68,11 +68,13 @@ def base_uri(aura_env: Optional[str] = None) -> str:
...
-    def create_attached_session(self, name: str, dbid: str, pwd: str, memory: SessionMemoryValue) -> SessionDetails:
-        return self._create_session(name, pwd, memory, instance_id=dbid)
+    def create_attached_session(
+        self, name: str, dbid: str, pwd: str, memory: SessionMemoryValue, ttl: Optional[str]
+    ) -> SessionDetails:
+        return self._create_session(name, pwd, memory, instance_id=dbid, ttl=ttl)
 
     def create_standalone_session(
-        self, name: str, pwd: str, memory: SessionMemoryValue, cloud_location: CloudLocation
+        self, name: str, pwd: str, memory: SessionMemoryValue, ttl: Optional[str], cloud_location: CloudLocation
```
Comment (FlorentinD) [inline graphdatascience/session/aura_api.py:77]: similar to memory, we could pass the ttl as the typed timedelta here

File: graphdatascience/session/dedicated_sessions.py [hunk 4]
```diff
@@ -55,10 +56,11 @@ def get_or_create(
             self._check_expiry_date(existing_session)
             self._check_memory_configuration(existing_session, memory.value)
             self._check_dbid(existing_session, dbid)
+            # TODO check cloud location?
```
Comment (FlorentinD) [inline graphdatascience/session/dedicated_sessions.py:59]: will be covered in https://trello.com/c/8oUk8poI

File: graphdatascience/session/gds_sessions.py [hunk 5]
```diff
@@ -89,12 +90,13 @@ def get_or_create(
...
             memory (SessionMemory): The size of the session specified by memory.
             db_connection (DbmsConnectionInfo): The database connection information.
+            ttl: Optional[timedelta]: The sessions time to live after inactivity in seconds.
             cloud_location (Optional[CloudLocation]): The cloud location. Required if the GDS session is for a self-managed database.
 
         Returns:
             AuraGraphDataScience: The session.
         """
-        return self._impl.get_or_create(session_name, memory, db_connection, cloud_location=cloud_location)
+        return self._impl.get_or_create(session_name, memory, db_connection, ttl=ttl, cloud_location=cloud_location)
```
Comment (FlorentinD) [inline graphdatascience/session/gds_sessions.py:99]: should throw if its used for AuraDS based sessions

**Review by FlorentinD — APPROVED** [review]
Looks good.

Main suggestion is to add it to the notebooks as well for user visibility (and CI testing)

**Review by Mats-SX — CHANGES_REQUESTED** [review]
I think we should define what activities refresh the TTL. But other than that, looks very good.

File: doc/modules/ROOT/pages/gds-session.adoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (DarthMax) [reply doc/modules/ROOT/pages/gds-session.adoc:90]: We are currently not describing the example at all. I think that if we add this line, we would also have to describe the rest of the parameters, which would be a duplication of the description above.

**Pull request merged**